model_utilities/translate/translator.py

```diff
@@ -23,8 +23,13 @@ def __init__(self, logger: Logger, model: AutoModelForSeq2SeqLM, tokenizer: Nllb
                                    tokenizer=self.__tokenizer,
                                    max_length=self.__max_length)
 
-    def translate(self, text: str, source_lang: str, target_lang: str) -> str:
+    def translate(self, text: list[str], source_lang: str, target_lang: str) -> list[str]:
         model_output = self.__pipeline(text, src_lang=source_lang, tgt_lang=target_lang)
-        translation_text = model_output[0]["translation_text"]
-        self.__logger.debug(f"Translation: '{text}' ({source_lang}) → '{translation_text}' ({target_lang})")
-        return translation_text
+
+        translation_text_list = []
+        for text_ in text:
+            translation_text = model_output[0]["translation_text"]
```

Comment:
You can't just take the zero'th index of the output for every sentence. Right now you're returning the same sentence multiple times.

Iterate over `model_output` and append `["translation_text"]` of every element.